execution/aggregate/accumulator.go

```diff
@@ -19,14 +26,14 @@ const (
 )
 
 type accumulator interface {
-	Add(v float64, h *histogram.FloatHistogram) error
+	Add(ctx context.Context, v float64, h *histogram.FloatHistogram) error
 	Value() (float64, *histogram.FloatHistogram)
 	ValueType() ValueType
 	Reset(float64)
 }
 
 type vectorAccumulator interface {
-	AddVector(vs []float64, hs []*histogram.FloatHistogram) error
+	AddVector(ctx context.Context, vs []float64, hs []*histogram.FloatHistogram) error
```

Comment:
It seems that the context is used is only used to store warnings. My instinct would be to return those warnings as values instead, and operators can add them to the context.